api.js

```diff
@@ -40,6 +40,7 @@ class Api {
     this._strictMode = 'strictMode' in opts ? opts.strictMode : false
     this._magicCommandAdded = false
     this._modulesSeen = opts.modulesSeen || []
+    this._errorFormatter = undefined
```

Comment:
On another note, when I first saw this, I thought the `_errorFormatter` property should be propagated down to child Api instances via `newChild`, but then I realized that it's only applied at the top-level since that's where the single Context instance (per parse) is initialized.

When documenting this, we should note that both `.unexpectedErrorFormatter(fn)` _and_ the `unexpectedError` style hook **can only be applied at the top-level** (i.e. cannot be applied/overridden by command configuration), because a single "top-level" Context instance is used for all (potentially nested) command levels.

Reply:
In preparation for implementing #65, perhaps we could go ahead and throw an error from `.unexpectedErrorFormatter(fn)` if it's called on a child Api instance?

Just trying to think in terms of setting a precedent for the future. Otherwise, I'm completely fine if the calls simply get ignored in the initial implementation of this feature.

Reply:
My preference is to start with no error for now -- in my opinion, if people complained about being bitten by this bug, then the "right fix" isn't to prevent people from passing the error formatter to subcommands, it's to fix it so it works as the user was assuming.

(That's out of scope for this PR, but I think it's possible if we cared enough -- we'd need to capture an unexpected error thrown at the `Command` level and capture the error formatter from that context, and then ensure that one got used if and when an unexpected error was printed.  I consider this a "potential future todo" that's better than just preventing people from using the option.)